### PR TITLE
fix: use correct node name as args to PodName. Fixes #9906

### LINKED ISSUE #9906: Wrong Pod name in `argo get` command result from CLI
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

Run an example workflow https://github.com/argoproj/argo-workflows/blob/master/examples/retry-on-error.yaml

The Pod name in the result of argo get are wrong.

argo get retry-on-error-v2pk2 -n workflow
Name:                retry-on-error-v2pk2
...

STEP                          TEMPLATE         PODNAME                                          DURATION  MESSAGE
 ✖ retry-on-error-v2pk2       error-container                                                             No more retries left
 ├─⚠ retry-on-error-v2pk2(0)  error-container  retry-on-error-v2pk2-error-container-2869263017  26s       Error (exit code 1): failed to put file: 404 Not Found
 ├─✖ retry-on-error-v2pk2(1)  error-container  retry-on-error-v2pk2-error-container-2427568992  4s        Error (exit code 3)
 └─✖ retry-on-error-v2pk2(2)  error-container  retry-on-error-v2pk2-error-container-816476283   4s        Error (exit code 4)

kubectl get pods -n workflow
NAME                                              READY   STATUS      RESTARTS   AGE
retry-on-error-v2pk2-error-container-1195955417   0/2     Completed   0          6m17s
retry-on-error-v2pk2-error-container-1800096796   0/2     Error       0          5m41s
retry-on-error-v2pk2-error-container-3410203767   0/2     Error       0          5m31s

The UI works good 
NAME
retry-on-error-v2pk2(0)   
ID
retry-on-error-v2pk2-1195955417   
POD NAME
retry-on-error-v2pk2-error-container-1195955417   

### Version

v3.4.1

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: retry-on-error-
spec:
  entrypoint: error-container
  templates:
  - name: error-container
    retryStrategy:
      limit: "2"
      retryPolicy: "Always"   # Retry on errors AND failures. Also available: "OnFailure" (default), "OnError", and "OnTransientError" (retry only on transient errors such as i/o or TLS handshake timeout. Available after v3.0.0-rc2)
    container:
      image: python
      command: ["python", "-c"]
      # fail with a 80% probability
      args: ["import random; import sys; exit_code = random.choice(range(0, 5)); sys.exit(exit_code)"]
```

### Logs from the workflow controller

Not related

### Logs from in your workflow's wait container

Not related

### PR DESCRIPTION
**Pull request by isubasinghe**
Signed-off-by: Isitha Subasinghe <isitha@pipekit.io>

Fixes #9906

Please do not open a pull request until you have checked ALL of these:

* [x] Create the PR as draft .
* [x] Run `make pre-commit -B` to fix codegen and lint problems.
* [x] Sign-off your commits (otherwise the DCO check will fail).
* [x] Use [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) (otherwise the commit message check will fail).
* [x] "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
* [x] Add unit or e2e tests. Say how you tested your changes. If you changed the UI, attach screenshots.
* [x] Github checks are green.
* [x] Once required tests have passed, mark your PR "Ready for review".

If changes were requested, and you've made them, dismiss the review to get it reviewed again.

### CODE REVIEW

**Comment by isubasinghe** [comment]
The issue was with the PodName function, the name doesn't suggest this but it generates names for you using a hash of the fields. 
PodName was called with the name+formatting applied to it, resulting in a different Hash. 

This is the simplest fix, but I personally think it would be nicer if two functions existed. A `GeneratePodName` and `GetPodName`. This way the generation is more obvious.

File: cmd/argo/commands/common/get.go [hunk 1]
```diff
@@ -479,7 +479,7 @@ func printNode(w *tabwriter.Writer, node wfv1.NodeStatus, wfName, nodePrefix str
 	var args []interface{}
 	duration := humanize.RelativeDurationShort(node.StartedAt.Time, node.FinishedAt.Time)
 	if node.Type == wfv1.NodeTypePod {
-		podName := util.PodName(wfName, nodeName, templateName, node.ID, podNameVersion)
+		podName := util.PodName(wfName, node.Name, templateName, node.ID, podNameVersion)
```
Comment (isubasinghe) [inline cmd/argo/commands/common/get.go:482]: nodeName was formatted with the emojis etc to display that it was success full / not. 
Using util.PodName then computes the hash based off of this, resulting in the seemingly random numbers. 

This is also why the `argo get -o json/yaml ` output is totally fine

Comment (isubasinghe) [reply cmd/argo/commands/common/get.go:482]: I am personally of the opinion `PodName` is a bit misleading and should be renamed `GeneratePodName`, this implies the (potential) non-determinism better.

**Comment by isubasinghe** [comment]
image
Here is the working `argo get` command

File: cmd/argo/commands/common/get.go (hunk 1 above)
Comment (juliev0) [reply cmd/argo/commands/common/get.go:482]: feel free to make that change if you like :)

Comment (juliev0) [inline cmd/argo/commands/common/get.go:482]: This is great. Do you think we should add a comment in here to explain the difference between "nodeName" and "node.Name"?

**Review by juliev0 — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]

**Comment by mweibel** [comment]
is there a possibility to release this as a bugfix soon?

**Comment by caelan-io** [comment]
@mweibel yes, we're looking at a new 3.4.4 patch release in the near future.

**Pull request merged**